src/java.net.http/share/classes/jdk/internal/net/http/HttpClientImpl.java

```diff
@@ -627,7 +638,7 @@ final long unreference() {
         final long httpCount = pendingHttpOperationsCount.decrementAndGet();
         final long http2Count = pendingHttp2StreamCount.get();
         final long webSocketCount = pendingWebSocketCount.get();
-        if (count == 0 && facade() == null) {
+        if (count == 0 && facadeRef.refersTo(null)) {
             selmgr.wakeupSelector();
```

Comment:
I notice there is only one remaining reference to the facade() method. It might make sense to get rid of it and replace the call site with `facadeRef.refersTo`

Reply:
The only remaining call I see is here:
```
        return new BuilderImpl(this.facade(), proxySelector);
```
and that call needs to return the facade - we can't replace it with `refersTo`. I'm not seeing any other calls.